### PR TITLE
Ensure we protect Collections obtained from scripts from self-referencing

### PR DESCRIPTION
**Pull request by s1monw**
Self referencing maps can cause SOE if they are iterated ie. in their toString methods. This chance adds some protected to the usage of those collections.

### CODE REVIEW

**Review by jasontedor — APPROVED** [review]
LGTM. Thanks @s1monw.

File: server/src/main/java/org/elasticsearch/search/aggregations/pipeline/bucketscript/BucketScriptPipelineAggregator.java
Comment (spinscale) [inline server/src/main/java/org/elasticsearch/search/aggregations/pipeline/bucketscript/BucketScriptPipelineAggregator.java]: shouldnt this be `CollectionUtils.ensureNoSelfReferences(returned);` ?

**Review by spinscale — COMMENTED** [review]
left one comment that needs fixing IMO, LGTM otherwise when tests are green

**Comment by s1monw** [comment]
@spinscale pushed a new commit

**Review by spinscale — APPROVED** [review]
LGTM

**Pull request merged**
